src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -34,13 +34,17 @@
 
 #include <App/DocumentObjectGroup.h>
 #include <App/OriginFeature.h>
+#include <Gui/Action.h>
+#include <Gui/BitmapFactory.h>
 #include <Gui/Application.h>
 #include <Gui/Document.h>
 #include <Gui/CommandT.h>
 #include <Gui/Control.h>
 #include <Gui/MainWindow.h>
 #include <Gui/SelectionFilter.h>
 #include <Gui/SelectionObject.h>
+#include <Gui/PrefWidgets.h>
+#include <QWidgetAction>

```

Comment:
QWidgetAction is a 3rd party header and thus should go to the section
```
#ifndef _PreComp_
#endif
```
There you also have to add `#include <QHBoxLayout>` to fix the build failure.

Reply:
Thanks for the comment.
I think I actually first tried to put QWidgetAction there, but it wouldn't work, VS was giving me an error. I'll try again see if I can find out why.

Got it for QHBoxLayout. For this one VS doesn't say anything and happily compiles. I need to find if he gives a warning somewhere because I had such issue several times already.

Reply:
It seems QWidgetAction is missing in QtAll.h

Reply:
So I leave it like this? Or is QtAll.h a FC file that needs fixing?